### PR TITLE
refactor(esl-share): merge list and group attributes

### PR DESCRIPTION
**Pull request by dshovchko**

### CODE REVIEW

**Review by ala-n — APPROVED** [review]

File: src/modules/esl-share/core/esl-share-list.ts [hunk 1]
```diff
@@ -56,33 +74,11 @@ export class ESLShareList extends ESLBaseElement {
 
   public get buttonsConfig(): Promise<ShareButtonConfig[]> {
     return (this.constructor as typeof ESLShareList).config().then((config) => {
-      const list = this.getList(config);
-      return list.length ? this.getButtons(config, list) : config.buttons;
+      const buttonsList = getButtonsList(config, this.list);
+      return buttonsList.length ? buttonsList : config.buttons;
     });
```
[the remark this answers is not in the document]
Comment (dshovchko) [reply src/modules/esl-share/core/esl-share-list.ts]: Sorry, missed it :(

fixed

**Review by fshovchko — APPROVED** [review]

**Review by abarmina — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

**Review by ala-n — APPROVED** [review]

**Review by yadamskaya — APPROVED** [review]

File: pages/views/examples/share.njk [hunk 2]
```diff
@@ -21,38 +21,48 @@ aside:
...
+  <esl-share-list list="kakao pusha alladin mix myspace"></esl-share-list>
   <hr>
 
   <h3>List of share buttons which presents buttons with all action types</h3>
   <p>This example shows a button with the native share mechanism on the device which will be inactive on the desktop browser.</p>
-  <esl-share-list list="copy,linkedin,mail,native-share,print"></esl-share-list>
+  <esl-share-list list="copy linkedin mail native-share print"></esl-share-list>
+  <hr>
+
+  <h3>Display list of share buttons with empty list</h3>
```
Comment (julia-murashko) [inline pages/views/examples/share.njk:54]:
```suggestion
  <h3>Displays list of share buttons with empty list</h3>
```

File: pages/views/examples/share.njk [hunk 3]
```diff
@@ -21,38 +21,48 @@ aside:
...
+  <p>Shows empty component without buttons.</p>
+  <esl-share-list list=""></esl-share-list>
   <hr>
 
   <h3>Display list of share buttons via incorrect list</h3>
```
Comment (julia-murashko) [inline pages/views/examples/share.njk:59]:
```suggestion
  <h3>Displays list of share buttons via incorrect list</h3>
```

File: pages/views/examples/share.njk [hunk 4]
```diff
@@ -21,38 +21,48 @@ aside:
...
-  <p>Shows buttons only with the correct id.</p>
-  <esl-share-list list="twittor , copi"></esl-share-list>
+  <p>Shows buttons only with the correct names.</p>
+  <esl-share-list list="twittor  copi"></esl-share-list>
   <hr>
 
   <h3>Display list of share buttons (without group or list)</h3>
```
Comment (julia-murashko) [inline pages/views/examples/share.njk:64]:
```suggestion
  <h3>Displays list of share buttons (without group or list)</h3>
```

**Review by julia-murashko — APPROVED** [review]

**Pull request merged**